### DISCUSSION TITLE
In DDP, how can I get that real batch_idx? since I found multiple repeated idx.

### DISCUSSION THREAD

**Discussion by Luciennnnnnn** [Lightning-AI/pytorch-lightning#6738] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
For example, when I use 4 GPU in DDP, then there are **4 same batch_idx is 0,** and this is not I wanted, since I will use this batch_idx as image name to write on desk.
How can I do that get **unique ordered idx** in `test_step`

**Comment by Luciennnnnnn** [comment]
Need help.

**Reply by ananthsub** [reply]
Could you clarify what you mean by "real" batch index? The indices here are local to the trainer rank. You could include the trainer rank and batch index to deduplicate. Alternatively your batch could contain the identifier that could use for the file name.

**Reply by Luciennnnnnn** [reply]
> Could you clarify what you mean by "real" batch index? The indices here are local to the trainer rank

Yes, for example, when I have a dataset consistent of 100 images, I will use `batch_size = 1` and `shuffle = False` in testing, so that I can save output image of my network **use the unique ordered idx (0 - 99)**
